File: Asm403_hoangvhph29660/model/userModel.js
const { default: mongoose } = require('mongoose');
var db = require('./db');

var userSchema = db.mongoose.Schema(
    {
        avata: { type: String },
        username: { type: String, required: true },
        password: { type: String, required: true },
        email: { type: String, required: true },
        fullname: { type: String, required: true },
        role: { type: String, required: true },
    },
    {
        collection: 'table_user'
    }
);

let user_model = db.mongoose.model("users_model", userSchema);
module.exports = { user_model };

File: asmMob403_client/app/src/main/java/hoangvhph29660/fpt/edu/asmmob403_client/model/UserModel.java
package hoangvhph29660.fpt.edu.asmmob403_client.model;

import java.io.Serializable;

public class UserModel implements Serializable {
    String _id;
    String avata , username , password , email , fullname , role;


    public UserModel(String avata, String username, String password, String email, String fullname, String role) {
        this.avata = avata;
        this.username = username;
        this.password = password;
        this.email = email;
        this.fullname = fullname;
        this.role = role;
    }

    public UserModel(String username, String password) {
        this.username = username;
        this.password = password;
    }

    public UserModel() {
    }

    public String get_id() {
        return _id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public String getAvata() {
        return avata;
    }

    public void setAvata(String avata) {
        this.avata = avata;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getFullname() {
        return fullname;
    }

    public void setFullname(String fullname) {
        this.fullname = fullname;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }
}
